cmd/antrea-agent/agent.go

```diff
@@ -864,6 +865,14 @@ func run(o *Options) error {
 			return fmt.Errorf("failed to connect uplink to OVS bridge: %w", err)
 		}
 	}
+	// secondaryNetworkController Initialize must be run after FlowRestoreComplete.
+	if features.DefaultFeatureGate.Enabled(features.SecondaryNetwork) {
```

Comment:
Why we must separate secondaryNetworkController.Initialize() and secondarynetwork.NewController()? Could not we just new the controller and initialize here together, so no change of existing code.

Reply:
secondaryNetworkController.PodController has dependency on "localPodInformer" (an instance of `cache.SharedIndexInformer`) , so PodController must be created before `localPodInformer.Get().Run()`; In the meanwhile, secondaryNetworkController.Initialize has attached uplink to OVS bridge, it must perform after the FlowRetoreWait option is removed on the OVS bridge. If we keep the Controller construct and Initialize function together, we need to think about the above two dependencies at the same time,  e.g., we shall also modify the order of `localPodInformer.Run()`(), this is a bit tricky from the long-term maintenance perspective. 
Another reason is, the general solutions in agent code is to split a controller's "new" and "Run" methods as two steps, this change is to make secondaryController more consistent.